chrome/background.js

```diff
@@ -12,82 +12,98 @@ chrome.webNavigation.onCompleted.addListener(async () => {
   const currentHost = new URL(tab.url).hostname;
   const rules = elements[currentHost];
   if (!rules) return;
-
   for (const rule of rules) {
-    applyFunctions[`applyKind${rule.kind}`](tab, rule);
+    const {script, params} = applyFunctions[`applyKind${rule.kind}`](rule);
+      chrome.scripting.executeScript({
+        target: {tabId: tab.id}, 
+        args: [params],
+        func: script
+      });
   }
 });
 
-const applyKind1 = (tab, { selector, extraInfo }) => {
+const applyKind1 = ({ selector, extraInfo }) => {
   const toRemoveClasses = extraInfo
     .split(";")
     .map((klass) => `"${klass}"`)
     .join(",");
-
-  chrome.tabs.executeScript(tab.id, {
-    code: `document.querySelectorAll("${selector}").forEach(node => node.classList.remove(${toRemoveClasses}))`,
-  });
+  console.log(toRemoveClasses)
+  return {
+    script: ({selector, toRemoveClasses}) => document.querySelectorAll(selector).forEach(node => node.classList.remove(toRemoveClasses)),
+    params: {selector, toRemoveClasses}
+  }
 };
 
-const applyKind2 = (tab, { selector, extraInfo }) => {
+const applyKind2 = ({ selector, extraInfo }) => {
   const toAddClasses = extraInfo
     .split(";")
     .map((klass) => `"${klass}"`)
     .join(",");
 
-  chrome.tabs.executeScript(tab.id, {
-    code: `document.querySelectorAll("${selector}").forEach(node => node.classList.add(${toAddClasses}))`,
-  });
+  return {
+    script: ({selector, toAddClasses}) => document.querySelectorAll(selector).forEach(node => node.classList.add(toAddClasses)),
+    params: {selector, toAddClasses}
+  }
 };
 
-const applyKind3 = (tab, { selector, extraInfo }) => {
-  chrome.tabs.executeScript(tab.id, {
-    code: `document.querySelectorAll("${selector}").forEach(node => node.style.cssText += "${extraInfo}")`,
-  });
+const applyKind3 = ({ selector, extraInfo }) => {
+  return {
+    script: ({selector, extraInfo}) => document.querySelectorAll(selector).forEach(node => node.style.cssText += extraInfo),
+    params: {selector, extraInfo}
+  }
 };
 
-const applyKind4 = (tab, { selector }) => {
-  chrome.tabs.executeScript(tab.id, {
-    code: `document.querySelectorAll("${selector}").forEach(node => node.parentNode.removeChild(node))`,
-  });
+const applyKind4 = ({ selector }) => {
+  return {
+    script: ({selector}) => document.querySelectorAll(selector).forEach(node => node.parentNode.removeChild(node)),
+    params: {selector}
+  }
 };
 
-const applyKind5 = (tab, { extraInfo }) => {
-  const script = `(function (){ ${extraInfo} })();`;
-
-  chrome.tabs.executeScript(tab.id, {
-    code: `
-      const bRemoverNode = document.createElement("script");
-      const bRemoverData = document.createTextNode(\`${script}\`);
-      bRemoverNode.appendChild(bRemoverData);
-      document.querySelector("head").appendChild(bRemoverNode);
-    `,
-  });
+const applyKind5 = ({ extraInfo }) => {
+  return {script: ({extraInfo}) => {
+    document.documentElement.setAttribute('onreset', extraInfo);

```

Comment:
Is there a specific reason for this to be called `onreset`? seems a bit misleading. I guess it's also possible for us to still use the old code of appending to the header, is there any benefit on this over the other option?